cli/src/main/java/com/devonfw/tools/ide/context/AbstractIdeContext.java

```diff
@@ -177,14 +177,14 @@ public AbstractIdeContext(IdeLogLevel minLogLevel, Function<IdeLogLevel, IdeSubL
       }
       currentDir = getParentPath(currentDir);
     }
+
     // detection completed, initializing variables
-    setCwd(userDir, workspace, currentDir);
     Path ideRootPath = null;
     if (currentDir == null) {
       info(getMessageIdeHomeNotFound());
     } else {
       debug(getMessageIdeHomeFound());
-      ideRootPath = this.ideHome.getParent();
+      ideRootPath = currentDir.getParent();
```

Comment:
can be `null` here
```suggestion
      if (currentDir != null) {
        ideRootPath = currentDir.getParent();
      }
```

Reply:
The check for currentDir being null already happens in line 183 just above

Reply:
https://github.com/devonfw/IDEasy/blob/88385bc05d214fe5c166238ae86eba48bb18be8f/cli/src/main/java/com/devonfw/tools/ide/context/AbstractIdeContext.java#L183-L187

there is already a check for that a couple of lines before 😉

Reply:
Oh, sorry for the spam. That review was directly before our daily and I completed it in a hurry.
My mistake...